python/pyfunc-server/pyfuncserver/protocol/upi/server.py

```diff
@@ -0,0 +1,41 @@
+import logging
+from concurrent import futures
+
+from caraml.upi.v1 import upi_pb2_grpc, upi_pb2
+from grpc_reflection.v1alpha import reflection
+import grpc
+
+from pyfuncserver.config import Config
+from pyfuncserver.model.model import PyFuncModel
+
+
+class PredictionService(upi_pb2_grpc.UniversalPredictionServiceServicer):
+    def __init__(self, model: PyFuncModel):
+        if not model.ready:
+            model.load()
+        self._model = model
+
+    def PredictValues(self, request, context):
+        return self._model.upiv1_predict(request=request, context=context)
+
+
+class UPIServer:
+    def __init__(self, model: PyFuncModel, config: Config):
+        self._predict_service = PredictionService(model=model)
+        self._config = config
+
+    def start(self):
+        server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
```

Comment:
why 10? can we make it configurable?

Reply:
It's arbitrary for now, will revisit this again when multiprocessing support is added.